autoscaler/controllers/common/processors.go

```diff
@@ -30,3 +36,52 @@ func FilterAndSortProcessorsByOrderHint(processors *odigosv1.ProcessorList, coll
 
 	return filteredProcessors
 }
+
+func DeleteProcessorByType(ctx context.Context, c client.Client, processorType string, namespace string) error {
```

Comment:
```suggestion
func DeleteFirstProcessorByType(ctx context.Context, c client.Client, processorType string, namespace string) error {
```

This function will delete exactly one processor, but we potentially may have many